src/Model/Term.php

```diff
@@ -118,12 +183,12 @@ public static function insertFromTagFieldByItemId($itemid, $tags)
 					}
 				}
 
-				$tags[$match[1] . trim($match[3], ',.:;[]/\"?!')] = $match[2];
+				$tags[$match[2]] = $match[1] . trim($match[3], ',.:;[]/\"?!');
 			}
 		}
 
-		foreach ($tags as $tag => $link) {
-			if (substr(trim($tag), 0, 1) == '#') {
+		foreach ($tags as $link => $tag) {
```

Comment:
Was the change from ```$tags as $tag => $link``` to ```$tags as $link => $tag``` some kind of bugfix?

Reply:
Yep. If you had two tags with the same label, one of them would be overwritten. The correct unique key of this array is the tag link. You aren't supposed to have multiple tags pointing to the same URL, but you can have similar tag labels pointing to different URLs (like two accounts with the same display name on different servers).